magento/magento2
# Add Header (h1 - h6) tags to layout xml htmlTags Allowed types

### Description (*)
The point of this PR is simply to allow H1 to H6 tags to be used in the layout XML as htmlTags so that users and theme creators can set titles with proper header tags using the XML. As an example a creator create a H2 container with a `Magento\Framework\View\Element\Text` block inside it to easily create titles where they need. 

### Manual testing scenarios (*)

1.add this block inside any referenceContainer you want

            <container name="some-title" htmlTag="h5" htmlClass="title">
                <block class="Magento\Framework\View\Element\Text" name="some.section.title">
                    <arguments>
                        <argument name="text" xsi:type="string">Your title here</argument>
                    </arguments>
                </block>
            </container>

2. Clear Cache and Reload page and see that the Header tag is now working and does not throw a container type not allowed error

### Questions or comments
I am not sure if there was a reason these weren't included so there may be something I am not thinking about but it seemed like it could be a helpful addition to devs and creators. 
This is my first time contributing to M2 so please let me know if something can be done better or if I have missed anything.

There is a failure in the code sniffer for the Container.php I have changed but these failures, I think, are not introduced by me?

### Contribution checklist (*)
